partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/DiadicOperator.kt

```diff
@@ -348,116 +371,74 @@ internal abstract class DiadicOperator(
-        fillTable(PType.TINYINT, PType.INTERVAL_YM) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-        fillTable(PType.TINYINT, PType.INTERVAL_DT) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-
-        // SmallInt
-        fillTable(PType.INTERVAL_YM, PType.SMALLINT) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.INTERVAL_DT, PType.SMALLINT) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.SMALLINT, PType.INTERVAL_YM) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-        fillTable(PType.SMALLINT, PType.INTERVAL_DT) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-
-        // Integer
-        fillTable(PType.INTERVAL_YM, PType.INTEGER) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.INTERVAL_DT, PType.INTEGER) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.INTEGER, PType.INTERVAL_YM) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-        fillTable(PType.INTEGER, PType.INTERVAL_DT) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-
-        // BigInt
-        fillTable(PType.INTERVAL_YM, PType.BIGINT) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.INTERVAL_DT, PType.BIGINT) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.BIGINT, PType.INTERVAL_YM) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-        fillTable(PType.BIGINT, PType.INTERVAL_DT) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-
-        // Numeric
-        fillTable(PType.INTERVAL_YM, PType.NUMERIC) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.INTERVAL_DT, PType.NUMERIC) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.NUMERIC, PType.INTERVAL_YM) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-        fillTable(PType.NUMERIC, PType.INTERVAL_DT) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-
-        // Decimal
-        fillTable(PType.INTERVAL_YM, PType.DECIMAL) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.INTERVAL_DT, PType.DECIMAL) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.DECIMAL, PType.INTERVAL_YM) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-        fillTable(PType.DECIMAL, PType.INTERVAL_DT) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-
-        // REAL
-        fillTable(PType.INTERVAL_YM, PType.REAL) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.INTERVAL_DT, PType.REAL) { lhs, rhs -> getIntervalNumberInstance(lhs, rhs) }
-        fillTable(PType.REAL, PType.INTERVAL_YM) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
-        fillTable(PType.REAL, PType.INTERVAL_DT) { lhs, rhs -> getNumberIntervalInstance(lhs, rhs) }
+        SqlTypeFamily.NUMBER.members.forEach {
+            val number = it
```

Comment:
In Kotlin, you can just set the variable directly

```suggestion
        SqlTypeFamily.NUMBER.members.forEach { number ->
```